bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleCloudAPI.java

```diff
@@ -164,17 +167,34 @@ void setConfig(GoogleTTSConfig config) {
         }
     }
 
-    private Credentials createCredentials(String serviceAccountKey) throws IOException {
-        try (ByteArrayInputStream bis = new ByteArrayInputStream(serviceAccountKey.getBytes())) {
-            GoogleCredentials credential = GoogleCredentials.fromStream(bis)
-                    .createScoped(Collections.singleton(GCP_SCOPE));
-            return FixedCredentialsProvider.create(credential).getCredentials();
+    private boolean checkOnAuthCode() {
+        String authcode = config.authcode;
+        if (authcode != null && !authcode.isEmpty()) {
+            logger.debug("Trying to get access and refresh tokens.");
+            try {
+                oAuthService.getAccessTokenResponseByAuthorizationCode(authcode, GCP_REDIRECT_URI);
+                config.authcode = null;
```

Comment:
In bindings it is possible to clear the configuration after it has been used. e.g.

https://github.com/openhab/openhab2-addons/blob/3225c72c4cf366df6d1ab57fab1c9e66da0d000c/bundles/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/handler/InnogyBridgeHandler.java#L196-L198

If we like to do the same here we need a reference to the `ConfigAdmin` and change the value during runtime. Otherwise I would recommend to clear the configuration parameter manually. If one will not do that one will see an error message during the next initialization of this service.

Reply:
Solved with https://github.com/openhab/openhab2-addons/pull/6489/commits/a443d22f2bdb43e4ff4ac715482910418b713a1b